src/mon/PGMap.cc

```diff
@@ -2527,16 +2528,21 @@ void PGMap::get_health_checks(
         summary = "Degraded data redundancy: ";
         sev = HEALTH_WARN;
         break;
-      case DEGRADED_FULL:
-        health_code = "PG_DEGRADED_FULL";
-        summary = "Degraded data redundancy (low space): ";
-        sev = HEALTH_ERR;
+      case BACKFILL_FULL:
+        health_code = "PG_BACKFILL_FULL";
+        summary = "Low space blocking backfill (add storage if this doesn't resolve itself): ";
+        sev = HEALTH_WARN;
         break;
       case DAMAGED:
         health_code = "PG_DAMAGED";
         summary = "Possible data damage: ";
         sev = HEALTH_ERR;
         break;
+      case RECOVERY_FULL:
+        health_code = "PG_RECOVERY_FULL";
+        summary = "Full OSDs blocking recovery: ";
```

Comment:
I think ``Degraded data redundancy (full OSDs blocking recovery)`` might be a bit better since it makes it more clear what the impact is?

Reply:
@liewegas Recovery itself doesn't imply degradedness.  If PGs are also degraded, there will be a separate health line about that.  If an OSD is marked out, recovery could be migrating data as you used above for backfill.

Reply:
@liewegas 
Example output of down/out OSD requiring recovery while OSDs are full.

    health: HEALTH_ERR
            3 full osd(s)
            1 pool(s) full
            Degraded data redundancy: 600/1800 objects degraded (33.333%), 1 pg degraded, 1 pg undersized
            Full OSDs blocking recovery: 1 pg recovery_toofull

I don't think we want to to see degraded twice.

            Degraded data redundancy: 600/1800 objects degraded (33.333%), 1 pg degraded, 1 pg undersized
            Degraded data redundancy (full OSDs blocking recovery): 1 pg recovery_toofull

Reply:
yep, agree!